rpc/src/rpc_url.rs

```diff
@@ -179,7 +187,7 @@ mod test {
                     host: "127.0.0.1".to_string(),
```

Comment:
Could you please add a couple more test cases to `SUPPORTED_URLS` here? e.g. one with just a username, one with username and password.